libflux/flux/src/lib.rs

```diff
@@ -71,8 +75,13 @@ pub struct ErrorHandle {
     pub err: CString,
 }
 
-impl<T: 'static + error::Error> From<T> for Box<ErrorHandle> {
-    fn from(err: T) -> Self {
+impl From<anyhow::Error> for Box<ErrorHandle> {
+    fn from(mut err: anyhow::Error) -> Self {
+        (&mut err).into()
+    }
+}
+impl From<&mut anyhow::Error> for Box<ErrorHandle> {
```

Comment:
Is the `&mut anyhow::Error` conversion really needed?

Reply:
Yes because of how the errors are handled at the C FFI layer we end up with a `&mut anyhow::Error` that we need to capture so that Rust can drop it. See https://github.com/influxdata/flux/pull/4131/files#diff-4cd6520d44cd3f8bc7a508423e85e1a011da344bb34a1389a5a0aab459258e40R462